src/Voron/Util/Settings/PathSettingBase.cs

```diff
@@ -94,5 +94,25 @@ public static string ToFullPath(string inputPath, string baseDataDirFullPath)
 
             return Path.GetFullPath(result); // it will unify directory separators
         }
+
+        public static bool IsSubDirectory(string userPath, string rootPath)
+        {
+            var rootDirInfo = new DirectoryInfo(rootPath);
+            var userDirInfo = new DirectoryInfo(userPath);
+
+            if (userDirInfo.FullName == rootDirInfo.FullName)
+                return true;
+
+            while (userDirInfo.Parent != null)
+            {
+                if (userDirInfo.Parent.FullName == rootDirInfo.FullName)
```

Comment:
I wonder if we need to make this a case insensitive comparison (on Windows, at least?)